egs/wsj/s5/steps/nnet3/train_dnn.py

```diff
@@ -496,6 +499,12 @@ def Train(args, run_opts):
     else:
         egs_dir = args.egs_dir
 
+    if not (GetIvectorExtractorId(args.dir) ==  GetIvectorExtractorId(egs_dir+'/info')):
```

Comment:
For back compatibility, I think you should allow this to succeed if one of the two is None.